### DISCUSSION TITLE
resilience4j Circuit Breaker is not working with suspend function -  fallback function never get triggered

### DISCUSSION THREAD

**Discussion by shadikkhan** [resilience4j/resilience4j#2041] [Q&A]
I am using Spring Boot 2.X.X with Kotlin for Microservices development. I am using resilience4j for circuit breaker implementation and its working for non suspend function but when I am using suspend function (which has CircuitBreaker annotation) which is calling another suspend function inside of it(and has exception) then fallback function never get triggered.

```

    @CircuitBreaker(name = "ApiDataAcces", fallbackMethod = FALLBACK_METHOD_NAME)
    override suspend fun getApiDataAccessAsync(
        request: RequestModel,
        headers: HashMap<String, String>
    ): RestResponse {
        val instance = library.updateWebClientApiHandler(request.baseRequest, headers)
        val query = if (request.templateName?.trim().isNullOrEmpty()) {
            TEMPLATE_NAMES
        } else {
            request.templateName
        }
        return instance.GetAsync(
            "Notification",
            "?templates=${
                withContext(Dispatchers.IO) {
                    URLEncoder.encode(query, ENCODER_UTF8)
                }
            }",
            headers
        )
    }


```

If I will remove this `withContext` then it will call fallback function.

Not sure if this is an existing issue resilience or is there way to solve this.

I am posting here to get some help from Community.

Thank you.

**Comment by RobWin** [comment]
Hi,
Resilience4j Annotations and Spring AOP do not work with suspending functions yet.
Unfortunately you have to use resilience4j-kotlin and decorate functions manually at the moment.

**Reply by shadikkhan** [reply]
Thanks @RobWin for the confirmation. Is there any example for resilience4j-kotlin and decorate functions manually that I can follow?
